### PR TITLE
Added support for genepool backend

### PR DESCRIPTION
**Pull request by fw2568**
Development of genepool client API will continue in project https://github.com/eryph-org/dotnet-genepoolclient.
Therefore all model data for genesets and genes has been removed from this project. Also the eryph-packer app was moved to dotnet-genepoolclient. 
The RepositoryGenePool has been updated for support of **unauthenticated** access to our **staging** genepool.
Authenticated access has to be added later by allow to configure a API key in eryph.

As the new genepool backend requires that genesets cannot be changed, implemention for "latest" genesets has been updated to support fodder genes with configuration of catlets. Therefore, the breeding process in VMManagement has been updated to the BreedAndFeed concept, where catlets are first bred and then fed with the food provided by their parents and their own preferences.  

adds another part of #10

### CODE REVIEW

**Review by ChristopherMann — CHANGES_REQUESTED** [review]

File: src/core/src/Eryph.VmManagement/CatletConfigExtensions.cs [hunk 1]
```diff
@@ -1,21 +1,155 @@
...
+            var updatedConfig =
+                from drives in
+                    (breedConfig.Drives ?? Array.Empty<CatletDriveConfig>()).Map(
+                        drive => ExpandDriveConfig(genepoolReader, drive)
+                    ).Traverse(l => l.AsEnumerable())
+                    .Map(l => l.Flatten())
+                let fodder = breedConfig.Fodder ?? Array.Empty<FodderConfig>()
+                from expandedFodder in
+                    fodder.Where(x=>!x.Remove.GetValueOrDefault()).Map(
+                        f => ExpandFodderConfig(genepoolReader, f,
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/core/src/Eryph.VmManagement/CatletConfigExtensions.cs:35]: nope, see eryph-org/dotnet-configmodel#52

File: src/modules/src/Eryph.Modules.VmHostAgent/Genetics/RepositoryGenePool.cs [hunk 2]
```diff
@@ -30,156 +34,163 @@ public RepositoryGenePool(IHttpClientFactory httpClientFactory, ILogger log, IFi
...
+                _log.LogTrace("gene part {part} hash: {hashString}", messageName, hashString);
+
+                if (hashString != partHash)
+                {
+                    _log.LogInformation("gene part '{part}' hash mismatch. Actual hash: {hashString}",
+                        messageName,
+                        hashString);
+
+                    _fileSystem.FileDelete(partFile);
+                    throw new VerificationException($"Failed to verify hash of gene part '{messageName}'");
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/modules/src/Eryph.Modules.VmHostAgent/Genetics/RepositoryGenePool.cs:187]: replaced by HashVerificationException

File: src/core/test/Eryph.VmManagement.Test/Eryph.VmManagement.Test.csproj [hunk 3]
```diff
@@ -4,6 +4,7 @@
     <TargetFramework>net6.0-windows7.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
+    <LangVersion>latest</LangVersion>
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/core/test/Eryph.VmManagement.Test/Eryph.VmManagement.Test.csproj:7]: set to 11

File: src/modules/src/Eryph.Modules.VmHostAgent/Genetics/GenePoolBase.cs [hunk 4]
```diff
@@ -26,9 +27,9 @@ public abstract class GenePoolBase
         return (geneParts[0], geneParts[1].ToLowerInvariant());
     }
 
-    protected static GeneSetManifestData ReadGeneSetManifest(string json)
+    protected static GenesetTagManifestData ReadGeneSetManifest(string json)
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/modules/src/Eryph.Modules.VmHostAgent/Genetics/GenePoolBase.cs:30]: true

File: src/modules/src/Eryph.Modules.Controller/Compute/UpdateCatletSaga.cs [hunk 5]
```diff
@@ -146,27 +154,42 @@ public Task Handle(OperationTaskStatusEvent<PrepareGeneCommand> message)
...
-                                    catletDriveConfig.Source.Contains(response.RequestedGene))
-                                {
-                                    catletDriveConfig.Source =
-                                        catletDriveConfig.Source.Replace(response.RequestedGene, response.ResolvedGene);
-                                }
-
-                            }
-                        }
-                    }
+                    //if (Data.BreedConfig != null)
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/modules/src/Eryph.Modules.Controller/Compute/UpdateCatletSaga.cs:157]: nope

File: src/core/src/Eryph.VmManagement/VirtualMachine.cs [hunk 6]
```diff
@@ -151,8 +152,10 @@ select repRecreated
                     continue;
                 }
 
-                var parentPathName = parent.Replace('/', '\\');
-                var genesetManifestPath = Path.Combine(genepoolPath, parentPathName, "geneset.json");
+                var parentIdentifier = GeneSetIdentifier.ParseUnsafe(parent);
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/core/src/Eryph.VmManagement/VirtualMachine.cs]: replaced with ILocalGenePoolReader

File: src/modules/src/Eryph.Modules.Controller/Compute/UpdateCatletSaga.cs [hunk 7]
```diff
@@ -108,28 +108,36 @@ public Task Handle(OperationTaskStatusEvent<ValidateCatletConfigCommand> message
                         .Map(parentConfig => parentConfig?.Breed(Data.Config, Data.Config.Parent) ?? Data.Config)
                         .IfNone(Data.Config);
 
+                    var geneIdentifiers = new List<GeneIdentifier>();
+                    geneIdentifiers.AddRange(breedConfig?.Drives?
+                        .Select(x => x.Source)
+                        .Where(t => t != null && t.StartsWith("gene:"))
+                        .Select(x => x == null 
+                            ? throw new NullReferenceException() // could not happen
+                            : GeneIdentifier.ParseUnsafe(GeneType.Volume, x)) ?? Array.Empty<GeneIdentifier>());
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/modules/src/Eryph.Modules.Controller/Compute/UpdateCatletSaga.cs]: updated as suggested

File: src/modules/src/Eryph.Modules.Controller/Compute/UpdateCatletSaga.cs [hunk 8]
```diff
@@ -108,28 +108,36 @@ public Task Handle(OperationTaskStatusEvent<ValidateCatletConfigCommand> message
...
 
-                    var geneCommands = new List<(GeneType Type, string GeneName)>();
-                    geneCommands.AddRange((breedConfig?.Drives?
+                    geneIdentifiers.AddRange(breedConfig?.Fodder?
                         .Select(x => x.Source)
-                        .Where(t => t != null && t.StartsWith("gene:") && t.Split(':').Length >= 2)
-                        .Select(t => t?.Remove(0,"gene:".Length) ?? "") ?? Enumerable.Empty<string>()).Select(x=> (GeneType.Volume, x)));
+                        .Where(t => t != null && t.StartsWith("gene:"))
```
[the remark this answers is not in the document]
Comment (fw2568) [reply src/modules/src/Eryph.Modules.Controller/Compute/UpdateCatletSaga.cs]: see new version, will also fail now

**Review by ChristopherMann — APPROVED** [review]

**Pull request merged**